tests/models/soil/test_soil_model.py

```diff
@@ -198,17 +270,147 @@ def test_generate_soil_model(
         assert (
             model.next_update
-            == datetime64(config["core"]["timing"]["start_time"]) + time_interval
+            == np.datetime64(config["core"]["timing"]["start_time"]) + time_interval
         )
-        # Run the update step and check that next_update has incremented properly
-        model.update()
-        assert (
-            model.next_update
-            == datetime64(config["core"]["timing"]["start_time"]) + 2 * time_interval
-        )
-        # Check that updates to data fixture are correct
-        assert allclose(dummy_carbon_data["mineral_associated_om"], end_carbon[0])
-        assert allclose(dummy_carbon_data["low_molecular_weight_c"], end_carbon[1])
 
     # Final check that expected logging entries are produced
     log_check(caplog, expected_log_entries)
+
+
+# Check that mocked function is called
+def test_update(mocker, soil_model_fixture, dummy_carbon_data):
+    """Test to check that the update step works and increments the update step."""
+
+    end_lmwc = [0.04980117, 0.01999411, 0.09992829, 0.00499986]
+    end_maom = [2.50019883, 1.70000589, 4.50007171, 0.50000014]
+
+    mock_integrate = mocker.patch(
+        "virtual_rainforest.models.soil.soil_model.SoilModel.integrate_soil_model"
+    )
+    mock_integrate.return_value = Dataset(
+        data_vars=dict(
+            new_lmwc=DataArray(end_lmwc, dims="cell_id"),
+            new_maom=DataArray(end_maom, dims="cell_id"),
+        )
+    )
+
+    soil_model_fixture.update()
+
+    # Check that integrator is called once (and once only)
+    mock_integrate.assert_called_once()
+
+    # Check that time has incremented correctly
+    assert soil_model_fixture.next_update == np.datetime64(
+        "2020-01-01"
+    ) + 2 * np.timedelta64(12, "h")
+
+    # Check that data fixture has been updated correctly
+    assert np.allclose(dummy_carbon_data["low_molecular_weight_c"], end_lmwc)
+    assert np.allclose(dummy_carbon_data["mineral_associated_om"], end_maom)
+
+
+def test_replace_soil_pools(dummy_carbon_data, soil_model_fixture):
+    """Test function to update soil pools."""
+
+    end_lmwc = [0.04980117, 0.01999411, 0.09992829, 0.00499986]
+    end_maom = [2.50019883, 1.70000589, 4.50007171, 0.50000014]
+
+    new_pools = Dataset(
+        data_vars=dict(
+            new_lmwc=DataArray(end_lmwc, dims="cell_id"),
+            new_maom=DataArray(end_maom, dims="cell_id"),
+        )
+    )
+
+    # Use this update to update the soil carbon pools
+    soil_model_fixture.replace_soil_pools(new_pools)
+
+    # Then check that pools are correctly incremented based on update
+    assert np.allclose(dummy_carbon_data["mineral_associated_om"], end_maom)
+    assert np.allclose(dummy_carbon_data["low_molecular_weight_c"], end_lmwc)
+
+
+@pytest.mark.parametrize(
+    argnames=["output", "raises", "expected_log"],
+    argvalues=[
+        pytest.param(
+            DotMap(
+                success=True,
```

Comment:
I don't see that we need `DotMap` here. You can just have the `argnames` as:

```python
argnames=["success", "y", "msg", "raises", "expected_log"],
```

And then have `y = None` when it fails and `msg  = None` when it succeeds.